src/nrfx/hal/nrf_clock.c

```diff
@@ -32,14 +32,62 @@ void nrf_clock_task_trigger(NRF_CLOCK_Type * p_reg, nrf_clock_task_t task)
 {
   if (task == NRF_CLOCK_TASK_HFCLKSTART) {
     NRF_CLOCK_regs.TASKS_HFCLKSTART = 1;
+    NRF_CLOCK_regs.HFCLKRUN = 1;
     nrf_clock_reqw_sideeffects_TASKS_HFCLKSTART();
   } else if (task == NRF_CLOCK_TASK_HFCLKSTOP) {
     NRF_CLOCK_regs.TASKS_HFCLKSTOP = 1;
     nrf_clock_reqw_sideeffects_TASKS_HFCLKSTOP();
   } else if (task == NRF_CLOCK_TASK_LFCLKSTART) {
     NRF_CLOCK_regs.TASKS_LFCLKSTART = 1;
+    NRF_CLOCK_regs.LFCLKRUN = 1;
     nrf_clock_reqw_sideeffects_TASKS_LFCLKSTART();
+  } else if (task == NRF_CLOCK_TASK_LFCLKSTOP) {
+    NRF_CLOCK_regs.TASKS_LFCLKSTOP = 1;
+    NRF_CLOCK_regs.LFCLKRUN = 0;
+  } else if (task == NRF_CLOCK_TASK_CAL) {
+    /*We don't want to model clock calibration as per now*/
+    NRF_CLOCK_regs.TASKS_CAL = 1;
```

Comment:
Without sideeffects these 3 ones are not going to do anything.
Note that setting the TASK register is a very phony thing in the models, which is immediately cleared by the HW model.

Reply:
Should I just remove writing to tasks?

Reply:
Depends on what is it that you are trying to do.
Why did you add these 3 conditions? (NRF_CLOCK_TASK_CAL, NRF_CLOCK_TASK_CTSTART and NRF_CLOCK_TASK_CTSTOP)
if they have not purpose, then better to not have the conditions, and have the error below be thrown to users so they know nothing happens.
If you cannot have that error, please create a stub in the NRF_CLOCK.c HW model file , which clears back the task register, and maybe prints a warning

Reply:
That warning is probably good enough. (Though I tend to prefer things to clearly blob up when a non implemented feature is used, unless it is a known missing feature)